.github/resources/check_flakes.py

```diff
@@ -0,0 +1,286 @@
+import argparse
+import logging
+from decimal import getcontext, Decimal, ROUND_UP
+from pathlib import Path
+from typing import Dict, Set
+
+from junitparser import JUnitXml, TestSuite
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import xml.etree.ElementTree as ET
+
+EWM_ALPHA = 1
+EWM_ADJUST = False
+HEATMAP_FIGSIZE = (100, 50)
+
+def parse_input_files(junit_files: str):
```

Comment:
I'd refactor these interdependent free functions into a class which takes the config in the ctor and has a publish public method to publish the markdown report. The rest can be protected, and most likely static